amgut/lib/data_access/sql_connection.py

```diff
@@ -1,69 +1,134 @@
-from __future__ import division
-
 # -----------------------------------------------------------------------------
-# Copyright (c) 2014--, The American Gut Development Team.
+# Copyright (c) 2014--, The Qiita Development Team.
 #
 # Distributed under the terms of the BSD 3-clause License.
 #
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
-
+from __future__ import division
 from contextlib import contextmanager
-from psycopg2 import connect, Error as PostgresError
+from itertools import chain
+from tempfile import mktemp
+
+from psycopg2 import connect, ProgrammingError, Error as PostgresError
 from psycopg2.extras import DictCursor
+from psycopg2.extensions import (
+    ISOLATION_LEVEL_AUTOCOMMIT, ISOLATION_LEVEL_READ_COMMITTED)
 
 from amgut.lib.config_manager import AMGUT_CONFIG
 
 
+def flatten(listOfLists):
+    # https://docs.python.org/2/library/itertools.html
+    return chain.from_iterable(listOfLists)
+
+
 class SQLConnectionHandler(object):
-    """Encapsulates the DB connection with the Postgres DB"""
-    def __init__(self, con=None):
+    """Encapsulates the DB connection with the Postgres DB
+
+    Parameters
+    ----------
+    admin : {'no_admin', 'no_database', 'database'}, optional
+        Whether or not to connect as the admin user. Options other than
+        `no_admin` depend on admin credentials in the qiita configuration. If
+        'admin_without_database', the connection will be made to the server
+        specified in the qiita configuration, but not to a specific database.
+        If 'admin_with_database', then a connection will be made to the server
+        and database specified in the qiita config.
+    """
+    def __init__(self, con=None, admin='no_admin'):
+        if admin not in ('no_admin', 'admin_with_database',
+                         'admin_without_database'):
+            raise RuntimeError("admin takes only {'no_admin', "
+                               "'admin_with_database', or "
+                               "'admin_without_database'}")
```

Comment:
The postgres section of the config (ag_config.txt.example) should be updated to contain these options (admin_user, admin_password)